### PR TITLE
{Release} Upgrade to Azure CLI 2.16.0

### PR DESCRIPTION
**Pull request by azclibot**
Updated Azure CLI version.

### CODE REVIEW

File: src/azure-cli/HISTORY.rst [hunk 1]
```diff
@@ -3,6 +3,97 @@
...
+
+**APIM**
+
+* `az apim restore`: New command to restore a backup of an API Management service (#15933)
+
+**App Service**
+
+* Bugfix #14857: Let users update webapp config even with access restriction (#15945)
+* `az functionapp create`: accept "--runtime python" and  "--runtime-version 3.9" as Azure Functions v3 parameter (#16020)
+* Fixes #16041: az webapp config ssl create results in unknown error (#16124)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:27]:
```suggestion
* Fix #16041: az webapp config ssl create results in unknown error (#16124)
```

File: src/azure-cli/HISTORY.rst [hunk 2]
```diff
@@ -3,6 +3,97 @@
...
+* Change MSI cluster to default (#16057)
+
+**APIM**
+
+* `az apim restore`: New command to restore a backup of an API Management service (#15933)
+
+**App Service**
+
+* Bugfix #14857: Let users update webapp config even with access restriction (#15945)
+* `az functionapp create`: accept "--runtime python" and  "--runtime-version 3.9" as Azure Functions v3 parameter (#16020)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:26]:
```suggestion
* `az functionapp create`: Accept "--runtime python" and  "--runtime-version 3.9" as Azure Functions v3 parameter (#16020)
```

File: src/azure-cli/HISTORY.rst [hunk 3]
```diff
@@ -3,6 +3,97 @@
...
+* Fixes #16041: az webapp config ssl create results in unknown error (#16124)
+
+**ARM**
+
+* `az deployment-scripts`: Removed preview flag (#16019)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:31]:
```suggestion
* `az deployment-scripts`: Remove preview flag (#16019)
```

File: src/azure-cli/HISTORY.rst [hunk 4]
```diff
@@ -3,6 +3,97 @@
...
+
+**Backup**
+
+* Fix #14976: CLI error improvements for ValueError and AttributeError cases (#15861)
+* `az backup protection undelete`: Added support for AzureWorkload protection undelete using CLI (#15979)
+* Fixed Bad Request Error for Correct Workload Type Input (#15999)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:37]:
```suggestion
* `az backup protection undelete`: Add support for AzureWorkload protection undelete using CLI (#15979)
* Fix Bad Request Error for Correct Workload Type Input (#15999)
```

File: src/azure-cli/HISTORY.rst [hunk 5]
```diff
@@ -3,6 +3,97 @@
...
+
+* Add SKU option (#15895)
+
+**Service Fabric**
+
+* Update SF app docs. only support for arm deployed resources (#16003)
+
+**Synapse**
+
+*  (#16095)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:95]:
```suggestion
*  Support synapse sql dw cmdlets and update az synapse workspace create cmdlet  (#16095)
```

**Review by jsntcy — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 6]
```diff
@@ -3,6 +3,97 @@
...
+2.16.0
+++++++
+
+**ACR**
+
+* Update description for KEK param (#15866)
+
+**AKS**
+
+* `az aks nodepool add/update/upgrade`: takes max surge parameter. (#15740)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:15]:
```suggestion
* `az aks nodepool add/update/upgrade`: Take max surge parameter. (#15740)
```

File: src/azure-cli/HISTORY.rst [hunk 7]
```diff
@@ -3,6 +3,97 @@
...
+* Add support for AGIC addon (#15993)
+* Change MSI cluster to default (#16057)
+
+**APIM**
+
+* `az apim restore`: New command to restore a backup of an API Management service (#15933)
+
+**App Service**
+
+* Bugfix #14857: Let users update webapp config even with access restriction (#15945)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:25]:
```suggestion
* Fix #14857: Let users update webapp config even with access restriction (#15945)
```

File: src/azure-cli/HISTORY.rst [hunk 8]
```diff
@@ -3,6 +3,97 @@
...
+
+* Add SKU option (#15895)
+
+**Service Fabric**
+
+* Update SF app docs. only support for arm deployed resources (#16003)
+
+**Synapse**
+
+*  Support synapse sql dw cmdlets and update az synapse workspace create cmdlet  (#16095)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:95]:
```suggestion
* Support synapse sql dw cmdlets and update az synapse workspace create cmdlet (#16095)
```

**Review by qwordy — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 9]
```diff
@@ -3,6 +3,97 @@
...
+
+* `az netappfiles account ad add`: Two optional parameters added, aes_encryption and ldap_signing (#15934)
+* `az netappfiles account backup-policy update`: Three optional parameters added named tags, type and id (#15934)
+* `az netappfiles snapshot policy create`: An optional parameter added named provisioning_state (#15934)
+
+**Network**
+
+* `az network network watcher configure`: Fix NetworkWatcherCountLimitReached error caused by case sensitivity of location value (#15801)
+* `az network application-gateway http-listener`: Fix bug that cannot create and update with WAF policy name (#15929)
+* Deprecate route table V1 (#16039)
```
Comment (haroldrandom) [inline src/azure-cli/HISTORY.rst:69]: Please consider whether to add more info to hint user what this history line to do what

**Review by haroldrandom — APPROVED** [review]

**Review by jiasli — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 10]
```diff
@@ -3,6 +3,97 @@
...
+2.16.0
+++++++
+
+**ACR**
+
+* Update description for KEK param (#15866)
+
+**AKS**
+
+* `az aks nodepool add/update/upgrade`: Take max surge parameter. (#15740)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:15]:
```suggestion
* `az aks nodepool add/update/upgrade`: Take max surge parameter (#15740)
```

File: src/azure-cli/HISTORY.rst [hunk 11]
```diff
@@ -3,6 +3,97 @@
...
+* Change MSI cluster to default (#16057)
+
+**APIM**
+
+* `az apim restore`: New command to restore a backup of an API Management service (#15933)
+
+**App Service**
+
+* Fix #14857: Let users update webapp config even with access restriction (#15945)
+* `az functionapp create`: Accept "--runtime python" and  "--runtime-version 3.9" as Azure Functions v3 parameter (#16020)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:26]:
```suggestion
* `az functionapp create`: Accept `--runtime python` and  `--runtime-version 3.9` as Azure Functions v3 parameter (#16020)
```

File: src/azure-cli/HISTORY.rst [hunk 12]
```diff
@@ -3,6 +3,97 @@
...
+* Fix Bad Request Error for Correct Workload Type Input (#15999)
+
+**CDN**
+
+* Add preview multi-origin support. (#15348)
+* Add BYOC auto-rotation. (#15348)
+
+**Key Vault**
+
+* Add a parameter `--include-managed` for `az keyvault key/secret list` to list managed resources (#15926)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:46]:
```suggestion
* `az keyvault key/secret list`: Add a parameter `--include-managed` to list managed resources (#15926)
```

**Review by kairu-ms — APPROVED** [review]

File: src/azure-cli/HISTORY.rst [hunk 13]
```diff
@@ -3,6 +3,97 @@
...
+**Monitor**
+
+* `az monitor metrics alert create`: Support dynamic thresholds for condition parameter. (#15820)
+* `az monitor metrics alert update`: Support dynamic thresholds for condition parameter. (#15820)
+* `az monitor metrics alert dimension create`: Build a metric alert rule dimension. (#15820)
+* `az monitor metrics alert condition create`: Build a metric alert rule condition. (#15820)
+
+**MySQL**
+
+* Add MySQL version upgrade cli (#15977)
```
Comment (zhoxing-ms) [inline src/azure-cli/HISTORY.rst:57]:
```suggestion
* Add MySQL version upgrade CLI (#15977)
```

File: src/azure-cli/HISTORY.rst (hunk 9 above)
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:69]:
```suggestion
* `az network route-table`: Deprecate route table V1 (#16039)
```

File: src/azure-cli/HISTORY.rst (hunk 10 above)
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:15]:
```suggestion
* `az aks nodepool add/update/upgrade`: Take max surge parameter (#15740)
```

File: src/azure-cli/HISTORY.rst [hunk 14]
```diff
@@ -3,6 +3,97 @@
...
+**Key Vault**
+
+* Add a parameter `--include-managed` for `az keyvault key/secret list` to list managed resources (#15926)
+
+**Monitor**
+
+* `az monitor metrics alert create`: Support dynamic thresholds for condition parameter. (#15820)
+* `az monitor metrics alert update`: Support dynamic thresholds for condition parameter. (#15820)
+* `az monitor metrics alert dimension create`: Build a metric alert rule dimension. (#15820)
+* `az monitor metrics alert condition create`: Build a metric alert rule condition. (#15820)
```
Comment (jsntcy) [inline src/azure-cli/HISTORY.rst:53]:
```suggestion
* `az monitor metrics alert create`: Support dynamic thresholds for condition parameter (#15820)
* `az monitor metrics alert update`: Support dynamic thresholds for condition parameter (#15820)
* `az monitor metrics alert dimension create`: Build a metric alert rule dimension (#15820)
* `az monitor metrics alert condition create`: Build a metric alert rule condition (#15820)
```

**Pull request merged**